worlds/_bizhawk/README.md

```diff
@@ -0,0 +1,279 @@
+
+```py
+from typing import TYPE_CHECKING
+
+from NetUtils import ClientStatus
+
+import worlds._bizhawk as bizhawk
+from worlds._bizhawk.client import BizHawkClient
+
+if TYPE_CHECKING:
+    from worlds._bizhawk.context import BizHawkClientContext
+
+
+class MyGameClient(BizHawkClient):
+    game = "My Game"
+    system = "GBA"
+    patch_suffix = ".apextension"
+
+    async def validate_rom(self, ctx: "BizHawkClientContext") -> bool:
+        try:
+            # Check ROM name/patch version
+            rom_name = ((await bizhawk.read(ctx.bizhawk_ctx, [(0x100, 6, "ROM")]))[0]).decode("ascii")
+            if rom_name != "MYGAME":
+                return False  # Not a MYGAME ROM
+        except bizhawk.RequestFailedError:
+            return False  # Not able to get a response, say no for now
+
+        # This is a MYGAME ROM
+        ctx.game = self.game
+        ctx.items_handling = 0b001
+        ctx.want_slot_data = True
+
+        return True
+
+    async def game_watcher(self, ctx: "BizHawkClientContext") -> None:
+        try:
+            # Read save data
+            save_data = await bizhawk.read(
+                ctx.bizhawk_ctx,
+                [(0x3000100, 20, "System Bus")]
+            )[0]
+
+            # Check locations
+            if save_data[2] & 0x04:
+                await ctx.send_msgs([{
+                    "cmd": "LocationChecks",
+                    "locations": [23]
+                }])
+
+            # Send game clear
+            if not ctx.finished_game and (save_data[5] & 0x01):
+                await ctx.send_msgs([{
+                    "cmd": "StatusUpdate",
+                    "status": ClientStatus.CLIENT_GOAL
+                }])
+
+        except bizhawk.RequestFailedError:
+            # The connector didn't respond. Exit handler and return to main loop to reconnect
+            pass
+```
+
+### Tips
+
+- Make sure your client gets imported when your world is imported. You probably don't need to actually use anything in
+your `client.py` elsewhere, but you still have to import the file for your client to register itself.
+- When it comes to performance, there are two directions to optimize:  
+  1. If you need to execute multiple commands on the same frame, do as little work as possible. Only read and write necessary data,
+  and if you have to use locks, unlock as soon as it's okay to advance frames. This is probably the obvious one.
+  2. Multiple things that don't have to happen on the same frame should be split up if they're likely to be slow.
+  Remember, the game watcher runs fewer times per second than you can count on your hands. Extra function calls on the
+  client aren't that big of a deal; the player will not notice if your `game_watcher` is slow. But the emulator has to
+  be done with any given set of commands in 1/60th of a second to avoid hiccups (faster still if your players use
+  speedup). Bundling multiple reads together is likelier to cause a worse user experience.
```

Comment:
Honestly, because of the "worse" later in the sentence, this could even just be "likely"
```suggestion
  speedup). Bundling multiple reads together is more likely to cause a worse user experience.
```

Reply:
Lol, yeah, this one is a bit of a mess. Rewrote most of it.